google-cloud-storage/src/main/java/com/google/cloud/storage/BucketInfo.java

```diff
@@ -1862,7 +1862,7 @@ public Rule apply(LifecycleRule lifecycleRule) {
               }));
     }
 
-    if (rules != null) {
+    if (!rules.isEmpty()) {
```

Comment:
Will this support rules deletion as well? I'm wondering if rules should be set to null at construction time and only have a non-null value when rules are returned.

Reply:
I'll test and update if necessary

Reply:
kokoro failed on this case

Reply:
Added some unit tests and refactored things afterward to account for clearing rules as well.